src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,167 @@ full_subcategory_inclusion _
 
 end truncated
 
+section epi_mono
+
+/-- A morphism in `simplex_category` is a monomorphism precisely when it is an injective function
+-/
+theorem mono_iff_injective {n m : simplex_category} {f: n ⟶ m} :
+  (category_theory.mono f) ↔ (function.injective ⇑f.to_preorder_hom) :=
+begin
+  split,
+  { intros m x y h,
+  set cx : [0] ⟶ n := const n x with def_cx,
+  set cy : [0] ⟶ n := const n y with def_cy,
+  have H : (cx ≫ f = cy ≫ f) := by {dsimp, rw h},
+  resetI, rw (cancel_mono f) at H,
+  have from_constant_function :
+    (∀ (z : fin(n.len + 1)), (z = (const n z).to_preorder_hom.to_fun 0)),
+  { intro z, simp, refl},
+  rw [from_constant_function x, from_constant_function y, ←def_cx, ←def_cy, H] },
```

Comment:
```suggestion
  { introsI m x y h,
    have H : const n x ≫ f = const n y ≫ f,
    { dsimp, rw h },
    change (n.const x).to_preorder_hom 0 = (n.const y).to_preorder_hom 0,
    rw cancel_mono f at H,
    rw H },
```
I golfed your proof a bit! A style point: after the `{`, we usually try to indent like I've done here. Also, `intros + resetI` can be combined into `introsI`.